opm/autodiff/StandardWellsDense_impl.hpp

```diff
@@ -187,7 +187,7 @@ namespace Opm {
         const double volume = 0.002831684659200; // 0.1 cu ft;
         for (int w = 0; w < nw; ++w) {
             bool allow_cf = allow_cross_flow(w, ebosSimulator);
-            const EvalWell bhp = getBhp(w);
+            const EvalWell& bhp = getBhp(w);
```

Comment:
Since the getBhp() method returns by value, what motivates this change? It should be correct as the returned temporary has extended scope to the lifetime of the reference, but is there a benefit to this style? If so, do you think it is a pattern we should follow in general? (And that is why I ask, primarily.) I do think there are a few minor potential drawbacks to it (code understanding and refactoring may be limited by not realizing we are dealing with a copy, not all will realize there is a temporary with extended lifetime etc.).